PolloPollo.Web/Controllers/ApplicationsController.cs

```diff
@@ -46,9 +108,25 @@ public void Put(int id, [FromBody] string value)
         }
 
         // DELETE: api/ApiWithActions/5
-        [HttpDelete("{id}")]
-        public void Delete(int id)
+        [Route("{userId}/{id}")]
+        [HttpDelete]
+        public async Task<bool> Delete(int userId, int id)
         {
+            var found = await _applicationRepository.FindAsync(id);
```

Comment:
UserId should be the same as in the claim.
If not the same then return Forbidden().

if yes then just call deleteAsync, and it should return the response from DeleteAsync.

Reply:
DeleteAsync could return a tuple to explain what went wrong, so Unauthorized() can be called if DeleteAsync fails due to not being the owner of an application.